bin/npg_run_is_deletable

```diff
@@ -1,278 +1,34 @@
-}
-
-exit $exit_status;
-
-#########
-# subroutines
-#
-sub time_limit_deletable {
-
-  my $delta_days = DateTime->now()->delta_days($run_status->date())->in_units('days');
-  if($delta_days >= $FS_DELETABLE_DAYS) {
-    $logger->warn(qq[time_limit: $id_run last status change was $delta_days days ago. Deletable]);
-    return 1;
-  }
-  $logger->warn(qq[time_limit: $id_run ($delta_days days ago) $id_run NOT deletable]);
-  return;
-}
-
-sub npg_tracking_deletable {
-  my $unconditional = shift;
-
-  my $crsd = $run_status->run_status_dict->description();
-  if ( $unconditional ) {
-    if (
-         ( any { $_ eq $crsd } @NPG_DELETABLE_UNCOND )
-           &&
-         ( $opts->{'ignore-time-limit'} || time_limit_deletable() )
-       ) {
-      $logger->warn(qq[npg_tracking: $id_run '$crsd'. Unconditionally deletable]);
-      return 1;
-    }
-    return;
-  }
-
-  if ( any { $_ eq $crsd } @NPG_DELETABLE_STATES ) {
-    $logger->warn(qq[npg_tracking: $id_run '$crsd'. Deletable]);
-    return 1;
-  }
-
-  $logger->warn(qq[npg_tracking: $id_run '$crsd' is NOT deletable]);
-  return;
-}
-
-sub irods_bam_deletable {
-
-  if ( npg_pipeline::validation::sequence_files
-      ->new( irods               => $irods,
-             logger              => $logger,
-             id_run              => $id_run,
-             runfolder_path      => $runfolder_path,
-             npg_tracking_schema => $s,
-             lims_driver_type    => $opts->{'lims-driver-type'} || undef,
-             collection          => $collection,
-             file_extension      => $file_extension)->archived_for_deletion() ) {
-
-    $logger->warn(qq[iRODS: run $id_run - sequence files archived. Deletable]);
-    return 1;
-  }
-
-  $logger->warn(qq[iRODS: run $id_run is NOT deletable]);
-  return;
-}
-
-sub autoqc_deletable {
-
-  if( npg_pipeline::validation::autoqc_files
-      ->new( irods          => $irods,
-             logger         => $logger,
-             id_run         => $id_run,
-             collection     => $collection,
-             file_extension => $file_extension,
-             skip_checks    => $opts->{'skip-autoqc-check'} )->fully_archived() ) {
-
-    $logger->warn(qq[Autoqc: run $id_run autoqc results fully archived. Deletable]);
-    return 1;
-  }
-
-  $logger->warn(qq[Autoqc: run $id_run is NOT deletable]);
-  return;
-}
-
-sub remove_staging_tag {
-
-  my $user = $ENV{'USER'};
-  $user = (!$user || ($user eq $PROD_USER_NAME)) ? $PROD_USER_DB_NAME : $user;
-  try {
-    $run_obj->unset_tag($user, 'staging');
-    $logger->warn("Staging tag is removed for run $id_run");
-  } catch {
-    $logger->error("Failed to remove staging tag: $_");
-  };
-  return;
-}
+$v->info(sprintf 'Running %s for run %i, runfolder %s',
+                 __FILE__, $v->id_run, $v->runfolder_path);
+
+my $deletable = $v->run();
+
+my $m = sprintf 'Run %i is %sdeletable.', $v->id_run, $deletable ? q[] : q[NOT ];
+$v->info($m);
+
+exit $deletable ? 0 : 1;
```

Comment:
```suggestion
exit ($deletable ? 0 : 1);
```
else precedence is such that  `exit $deleteable` is evaluated before the `?`

Reply:
(not $deletable) is logically correct, but returns an undefined valie in case $deletable is true. Is this acceptable for a script return value?

Reply:
If think by the time it gets to the system call it will be converted to integers in the range 0 to 127.
```bash
$ perl -le 'exit not 1'; echo $?
0
$ perl -le 'exit not 0'; echo $?                                                                                                                                                                
1
```

Reply:
Thank you, done